tuf/ngclient/updater.py

```diff
@@ -122,12 +131,80 @@ def refresh(self) -> None:
             RepositoryError: Metadata failed to verify in some way
             DownloadError: Download of a metadata file failed in some way
         """
+        ordered_version_paths = self._set_spec_version()
 
-        self._load_root()
+        self._load_root(ordered_version_paths)
         self._load_timestamp()
         self._load_snapshot()
         self._load_targets(Targets.type, Root.type)
 
+    def _set_spec_version(self) -> List[str]:
+        try:
+            version_bytes = self._load_local_metadata("spec_version")
+            self._spec_version = json.loads(version_bytes.decode("utf-8"))[
+                "version"
+            ]
+
+        except OSError as e:
+            if e.errno != errno.ENOENT:
+                raise
+            self._spec_version = "1"
+
+        repository_versions_and_paths = self._get_repository_versions()
+
+        repository_versions = [
```

Comment:
Nit, but still makes it more readable:
```suggestion
        repository_versions: List[str] = [
```